packages/client/src/base/action-dispatcher.ts

```diff
@@ -69,11 +69,11 @@ export class GLSPActionDispatcher extends ActionDispatcher {
                 this.timeouts.delete(action.responseId);
             }
 
-            // we might have reached a timeout, so we simply drop the response
+            // Check if we have a pending request for the response.
+            // If not the  we clear the responseId => action will be dispatched normally
```

Comment:
That is quite the change in behavior and I believe it needs to be documented somewhere :thinking:

Reply:
Will be added to the changelog (I will do a collective update before the 2.0.0 release).
Anyhow, it's more of a bug fix. With the previous implementation it was not possible to dispatch a `Response` action with a set resposne id when using the `GLSPActionDispatcher`. But this is possible (and also documented in the jsdoc of request & requestUntil) with the super implementation from sprotty.

(Note that the response id clearing is optional, and is more or less just there to avoid an unnecesary warning log in the super implentation)